### ISSUE TITLE
lightconv Segmentation fault(core dumped) after training finishes

### ISSUE THREAD

**Issue by kinalmehta** [pytorch/fairseq#1442]
**Environment Setup**
AWS instance: g4dn.xlarge
Ubuntu 18.04 Deep learning EC2 AMI
CUDA 10.0
miniconda
python 3.6.9
pytorch                   1.3.1           py3.6_cuda10.0.130_cudnn7.6.3_0
fairseq cloned and installed editable version

**Command Run**
 fairseq-train data-bin/iwslt14.tokenized.de-en --clip-norm 0 --optimizer adam --lr 0.0005     --source-lang de --target-lang en --max-tokens 4000 --no-progress-bar     --log-interval 100 --min-lr '1e-09' --weight-decay 0.0001     --criterion label_smoothed_cross_entropy --label-smoothing 0.1     --lr-scheduler inverse_sqrt     --ddp-backend=no_c10d     --max-update 1000 --warmup-updates 4000 --warmup-init-lr '1e-07'     --adam-betas '(0.9, 0.98)' --keep-last-epochs 10     -a lightconv_iwslt_de_en --save-dir $SAVE     --dropout 0.3 --attention-dropout 0.1 --weight-dropout 0.1     --encoder-glu 0 --decoder-glu 0 --save-dir $SAVE --fp16

**Ouput:**
| model lightconv_iwslt_de_en, criterion LabelSmoothedCrossEntropyCriterion
| num. model params: 38685184 (num. trained: 38685184)
| training on 1 GPUs
| max tokens per GPU = 4000 and max sentences per GPU = None
| loaded checkpoint save/dynamic_conv_iwslt/checkpoint_last.pt (epoch 1 @ 1000 updates)
| loading train data for epoch 1
| loaded 160239 examples from: data-bin/iwslt14.tokenized.de-en/train.de-en.de
| loaded 160239 examples from: data-bin/iwslt14.tokenized.de-en/train.de-en.en
| data-bin/iwslt14.tokenized.de-en train de-en 160239 examples
| epoch 001 | loss 8.118 | nll_loss 7.368 | ppl 165.24 | wps 13604 | ups 4 | wpb 3733.700 | bsz 176.800 | num_updates 1010 | lr 0.000126325 | gnorm 1.013 | clip 0.000 | oom 0.000 | loss_scale 32.000 | wall 3 | train_wall 115
| epoch 001 | valid on 'valid' subset | loss 8.134 | nll_loss 7.297 | ppl 157.29 | num_updates 1010 | best_loss 8.13445
| saved checkpoint save/dynamic_conv_iwslt/checkpoint_best.pt (epoch 1 @ 1010 updates) (writing took 5.244306325912476 seconds)
| done training in 9.0 seconds
Segmentation fault (core dumped)

I don't understand why this happens after the training is finished. There is nothing unusual visible in the GPU or CPU memory too.
